cpa/dbconnect.py

```diff
@@ -665,36 +665,59 @@ def GetPerImageObjectCounts(self):
         for r in result1:
             counts[r[:-1]] = r[-1]
         return [r+(counts[r],) for r in result2 if r in counts]
-    
+
     def GetAllImageKeys(self):
         ''' Returns a list of all image keys in the image_table. '''
         select = "SELECT "+UniqueImageClause()+" FROM "+p.image_table+" GROUP BY "+UniqueImageClause()
         return self.execute(select)
-    
+
     def GetObjectsFromImage(self, imKey):
         return self.execute('SELECT %s FROM %s WHERE %s'%(UniqueObjectClause(), p.object_table, GetWhereClauseForImages([imKey])))
-    
+
     def GetObjectCoords(self, obKey, none_ok=False, silent=False):
-        '''Returns the specified object's x, y coordinates in an image.
-        '''
-        res = self.execute('SELECT %s, %s FROM %s WHERE %s'%(
-                        p.cell_x_loc, p.cell_y_loc, p.object_table, 
-                        GetWhereClauseForObjects([obKey])), silent=silent)
+        '''Returns the specified object's x, y, z coordinates in an image.
+        '''
+        if p.cell_z_loc:
```

Comment:
I've changed this now to depend on a separate 3D processing variable (`process_3D`) rather than a z location being present. It's still an optional variable, but I've matched how optional variables like `use_legacy_fetcher` are defined and I believe if left undefined, they will default to False (at least for the example of `use_legacy_fetcher` and the new `process_3D` variable). Please let me know if my understanding of this is off.